satpy/tests/reader_tests/test_gerb_l2_hr_h5.py

```diff
@@ -0,0 +1,217 @@
+        fid['/Scene Identification/Aerosol Optical Depth IR 1.6'].attrs['Quantisation Factor'] = \
+            np.array(0.001, dtype='float64')
+        fid['/Scene Identification/Aerosol Optical Depth VIS 0.6'] = np.ones(shape=(1237, 1237), dtype=np.dtype('>i2'))
+        fid['/Scene Identification/Aerosol Optical Depth VIS 0.6'].attrs['Quantisation Factor'] = \
+            np.array(0.001, dtype='float64')
+        fid['/Scene Identification/Aerosol Optical Depth VIS 0.8'] = np.ones(shape=(1237, 1237), dtype=np.dtype('>i2'))
+        fid['/Scene Identification/Aerosol Optical Depth VIS 0.8'].attrs['Quantisation Factor'] = \
+            np.array(0.001, dtype='float64')
+        fid['/Scene Identification/Cloud Cover'] = np.ones(shape=(1237, 1237), dtype=np.dtype('uint8'))
+        fid['/Scene Identification/Cloud Cover'].attrs['Quantisation Factor'] = np.array(0.01, dtype='float64')
+        dt = h5py.h5t.TypeID.copy(h5py.h5t.C_S1)
+        dt.set_size(8)
+        dt.set_strpad(h5py.h5t.STR_NULLTERM)
+        at = h5py.h5a.create(fid['/Scene Identification/Cloud Cover'].id, b'Unit', dt, h5py.h5s.create(h5py.h5s.SCALAR))
+        at.write(np.array(b'Percent', dtype='|S8'))
+        fid['/Scene Identification/Cloud Optical Depth (logarithm)'] = \
+            np.ones(shape=(1237, 1237), dtype=np.dtype('>i2'))
+        fid['/Scene Identification/Cloud Optical Depth (logarithm)'].attrs['Quantisation Factor'] = \
+            np.array(0.00025, dtype='float64')
+        fid['/Scene Identification/Cloud Phase'] = np.ones(shape=(1237, 1237), dtype=np.dtype('uint8'))
+        fid['/Scene Identification/Cloud Phase'].attrs['Quantisation Factor'] = np.array(0.01, dtype='float64')
+        dt = h5py.h5t.TypeID.copy(h5py.h5t.C_S1)
+        dt.set_size(34)
+        dt.set_strpad(h5py.h5t.STR_NULLTERM)
+        at = h5py.h5a.create(fid['/Scene Identification/Cloud Phase'].id, b'Unit', dt, h5py.h5s.create(h5py.h5s.SCALAR))
+        at.write(np.array(b'Percent (Water=0%,Mixed,Ice=100%)', dtype='|S34'))
+        fid['/Scene Identification/Dust Detection'] = np.ones(shape=(1237, 1237), dtype=np.dtype('uint8'))
+        fid['/Scene Identification/Dust Detection'].attrs['Quantisation Factor'] = np.array(0.01, dtype='float64')
+        fid['/Scene Identification/Solar Angular Dependency Model'] = np.ones(shape=(1237, 1237), dtype=np.dtype('>i2'))
+        fid['/Scene Identification/Surface Type'] = np.ones(shape=(1237, 1237), dtype=np.dtype('uint8'))
+        fid['/Scene Identification/Thermal Angular Dependency Model'] = \
+            np.ones(shape=(1237, 1237), dtype=np.dtype('uint8'))
+        fid.create_group('/Times')
+        fid['/Times/Time (per row)'] = np.ones(shape=(1237,), dtype=np.dtype('|S22'))
+
+    return filename
+
+
+def test_gerb_solar_flux_dataset(gerb_l2_hr_h5_dummy_file):
+    """Test the GERB L2 HR HDF5 file.
+
+    Load the solar flux component.
+    """
```

Comment:
We already know we're testing the GERB reader based on the module we're in so you can be a little less detailed in your docstrings. Same goes for the `thermal_flux`.

```suggestion
    """Test loading the solar flux component."""
```

And not that you have to change it, but you could do some fancy stuff with pytest to run both of these tests in the same function:

```python
@pytest.mark.parametrize("name", ["Solar Flux", "Thermal Flux"])
def test_dataset_load(gerb_l2_hr_h5_dummy_file, name):
```

Reply:
Applied the later fix with co-authored commit